src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker.cs

```diff
@@ -361,9 +361,11 @@ private Task<ProjectInfo> CreateProjectInfoAsync(ProjectFileInfo projectFileInfo
                         .WithStrongNameProvider(new DesktopStrongNameProvider(commandLineArgs.KeyFileSearchPaths))
                         .WithAssemblyIdentityComparer(DesktopAssemblyIdentityComparer.Default);
 
-                    var documents = CreateDocumentInfos(projectFileInfo.Documents, projectId, commandLineArgs.Encoding);
-                    var additionalDocuments = CreateDocumentInfos(projectFileInfo.AdditionalDocuments, projectId, commandLineArgs.Encoding);
-                    var analyzerConfigDocuments = CreateDocumentInfos(projectFileInfo.AnalyzerConfigDocuments, projectId, commandLineArgs.Encoding);
+                    // Since we don't know the encoding of all documents up front when loading a Project. 
+                    // Passing `null` for the encoding causes the FileTextLoader to try and determine the encoding for each document individually.
+                    var documents = CreateDocumentInfos(projectFileInfo.Documents, projectId, encoding: null);
+                    var additionalDocuments = CreateDocumentInfos(projectFileInfo.AdditionalDocuments, projectId, encoding: null);
+                    var analyzerConfigDocuments = CreateDocumentInfos(projectFileInfo.AnalyzerConfigDocuments, projectId, encoding: null);
```

Comment:
The default encoding is null anyways. The problem is the auto-detect logic is detecting one specific encoding when multiple encodings are valid. In the event we encounter a file with a different encoding than expected, the formatter should write the content of the file to a byte array in memory with the expected encoding, and then compare that with the binary content of the original file. A warning should only be reported if the bytes are different.